src/main/java/uk/ac/cam/cl/dtg/segue/search/IsaacSearchInstructionBuilder.java

```diff
@@ -0,0 +1,368 @@
+            instruction.mustNot(new MatchInstruction(Constants.TAGS_FIELDNAME, HIDE_FROM_FILTER_TAG));
+        }
+
+        // Exclude deprecated content
+        instruction.mustNot(new MatchInstruction(Constants.DEPRECATED_FIELDNAME, "true"));
+
+        return instruction;
+    }
+
+    /**
+     * Include only content of {@code contentTypes} in results. Other content types will be excluded. Subsequent calls
+     * add to the set of included content types.
+     *
+     * @param contentTypes A list of content types to match.
+     * @return This IsaacSearchQueryBuilder, to allow chained operations.
+     */
+    public IsaacSearchInstructionBuilder includeContentTypes(final Set<String> contentTypes) {
+        return this.includeContentTypes(contentTypes, Priority.NORMAL);
+    }
+
+    /**
+     * Include only content of {@code contentTypes} in results. Other content types will be excluded. Subsequent calls
+     * add to the set of included content types.
+     *
+     * @param contentTypes A list of content types to match.
+     * @param priority The priority of matches for this content type.
+     * @return This IsaacSearchQueryBuilder, to allow chained operations.
+     */
+    public IsaacSearchInstructionBuilder includeContentTypes(final Set<String> contentTypes, final Priority priority) {
+        if (priority == Priority.HIGH) {
+            this.priorityIncludedContentTypes.addAll(contentTypes);
+        } else {
+            this.includedContentTypes.addAll(contentTypes);
+        }
+        return this;
+    }
+
+    /**
+     * @param searchInField A {@code SearchInField} instance describing terms to look for in a particular field, and
+     *                      what priority resultant matches will have in the results (in absence of any sorting applied
+     *                      elsewhere). If no search terms are defined for {@code searchInField} it is ignored.
+     *
+     * @return This IsaacSearchQueryBuilder, to allow chained operations.
+     */
+    public IsaacSearchInstructionBuilder searchFor(SearchInField searchInField) {
+        if (!searchInField.getTerms().isEmpty()) {
+            this.searchesInFields.add(searchInField);
+        }
+        return this;
+    }
+
+    /**
+     * Sets whether to return events where the date field contains a date in the past. Defaults to false, and has no
+     * effect if the event content type is excluded.
+     *
+     * @param includePastEvents Whether to include past events.
+     * @return This IsaacSearchQueryBuilder, to allow chained operations.
+     */
+    public IsaacSearchInstructionBuilder includePastEvents(final boolean includePastEvents) {
+        this.includePastEvents = includePastEvents;
+        return this;
+    }
+
+    /**
+     * Builds and returns the final BooleanMatchInstruction reflecting the builder's settings.
+     *
+     * @return A BooleanMatchInstruction reflecting the builder's settings.
+     */
+    public BooleanInstruction build() {
```

Comment:
I'd argue that this should not mutate `masterInstruction` but that might just be me. If you call this `n` times, you'll end up with a `BooleanInstruction` with `n` times the "shoulds" that you would expect.

On the other hand, if the intention is that this is the last method you can ever call on an `IsaacSearchInstructionBuilder`, it could be re-initialised at the end of this method or something.